### DISCUSSION TITLE
Customized prompts not clear to me

### DISCUSSION THREAD

**Discussion by Lovenoreus** [microsoft/call-center-ai#378] [Q&A] [answered]
I assume that after I have deployed the resources and synced the settings I am supposed edit config.yaml to contain this example value down bellow. I didn't know to if I should run "make deploy" or "make dev" after adding the prompt information so I tried both.

I recieve errors relating to how I am suppoed to put things in a different format but I have a hard time understanding, I used the format of your example and verified that it's correct yaml. Seems like when it is passed to different scripts it is not correct anymore.

I'm attaching my console errors as a log file.
[log.txt](https://github.com/user-attachments/files/18158112/log.txt)

prompts:
  tts:
    hello_tpl:
      - : |
        Hello, I'm Digge, I'm an IT support specialist.
  llm:
    default_system_tpl: |
      Assistant is called Digge and is in a call center for the company Example Healthcare. He is an IT support specialist.
    chat_system_tpl: |
      Try to be nice

**Comment by Lovenoreus** [comment]
The tabs were formatted correctly unlike in my post

I can run both commands perfectly fine while leaving the prompt property as its default value

**Reply by clemlesne** [reply, marked as answer]
The error in the logs directly point to bad YAML formatting.

Here is how to format a multiline string with YAML:

```yaml
test:
  abc: |
    example of
    multiline string
```

Your formatter in your IDE (VS Code for example) should warn you about errors.

Or, go to websites like https://jsonformatter.curiousconcept.com to lint your YAML.

**Reply by Lovenoreus** [reply]
Yes but it was two different errors, if I only use single lines in the yaml I still get an error when I do "make dev" directly after.

It is trying to make the config.yaml prompt into a list which it can't since it is not valid list-format

"Input should be a valid list [type=list_type, input_value="Hello, I'm Manuel, 'm an IT support specialist.", input_type=str]"

Attaching the console-log and the picture of the prompt value

![image](https://github.com/user-attachments/assets/3355a445-2ca2-4015-a19b-e979462d2e63)
[log_prompt_makedev.txt](https://github.com/user-attachments/files/18163021/log_prompt_makedev.txt)

**Reply by clemlesne** [reply]
Indeed, what you typed is not a list but a string. The configuration requires a list, like the error showed and the documentation noted.

**Reply by Lovenoreus** [reply]
I see, I didn't know about the yaml-list syntax but now mine looks correct and can run

When you change the prompt in the config.yaml to test out behaviours in local development, what do you need to do after? If I just run "make deploy" then I was able to change the behaviour of the bot in the cloud container but calls are no longer being sent to the local development code

**Comment by Lovenoreus** [comment]
EDIT: I was able to solve it by using a particular order of deployments and changing numbers

I noticed that the local dev version of the app does not recieve calls at all after doing a new "make deploy"

I followed our previous instructions and synced settings, then add a seperate phone number, but no infomation that a call was made reaches the local app. There is no event grid with a number corresponding to the one I changed to i my local config.yaml but I think that is expected.

I just changed the phone numbers in config.yaml then ran make tunnel&make dev

[2024-12-18 15:57:20 +0100] [135976] [INFO] Starting gunicorn 23.0.0
[2024-12-18 15:57:20 +0100] [135976] [INFO] Listening at: http://0.0.0.0:8080 (135976)
[2024-12-18 15:57:20 +0100] [135976] [INFO] Using worker: uvicorn.workers.UvicornWorker
[2024-12-18 15:57:20 +0100] [135977] [INFO] Booting worker with pid: 135977
[2024-12-18 15:57:20 +0100] [135978] [INFO] Booting worker with pid: 135978
Env file loaded from "/home/user/PycharmProjects/call-center-ai/.env"
Env file loaded from "/home/user/PycharmProjects/call-center-ai/.env"
Cannot find env "CONFIG_JSON", trying to load from file
Cannot find env "CONFIG_JSON", trying to load from file
Config loaded from file "/home/user/PycharmProjects/call-center-ai/config.yaml"
Config loaded from file "/home/user/PycharmProjects/call-center-ai/config.yaml"
2024-12-18T14:57:31.017757Z [info     ] Using Cosmos DB call-center-ai/calls-v3
2024-12-18T14:57:31.049889Z [info     ] Using Cosmos DB call-center-ai/calls-v3
2024-12-18T14:57:31.406089Z [info     ] Using Communication Services from number +4532727047

**Reply by clemlesne** [reply]
Ok, I take this comment as resolved

**Reply by clemlesne** [reply]
Related this issue to enhance the documentation of the dev install steps: https://github.com/microsoft/call-center-ai/issues/380.
